Makefile

```diff
@@ -293,36 +298,39 @@ ifeq ($(BUILDX),true)
 	./release-kong.sh
 endif
 
-test: setup-tests build-test-container
+test: build-test-container
 ifneq ($(RESTY_IMAGE_BASE),src)
+	VERBOSE=$(VERBOSE) \
 	KONG_VERSION=$(KONG_VERSION) \
 	RESTY_IMAGE_BASE=$(RESTY_IMAGE_BASE) \
 	RESTY_IMAGE_TAG=$(RESTY_IMAGE_TAG) \
 	KONG_PACKAGE_NAME=$(KONG_PACKAGE_NAME) \
 	KONG_TEST_CONTAINER_TAG=$(KONG_TEST_CONTAINER_TAG) \
-	KONG_TEST_CONTAINER_NAME=$(KONG_TEST_CONTAINER_NAME) \
+	KONG_TEST_IMAGE_NAME=$(KONG_TEST_IMAGE_NAME) \
 	RESTY_VERSION=$(RESTY_VERSION) \
 	RESTY_OPENSSL_VERSION=$(RESTY_OPENSSL_VERSION) \
 	RESTY_LUAROCKS_VERSION=$(RESTY_LUAROCKS_VERSION) \
 	RESTY_PCRE_VERSION=$(RESTY_PCRE_VERSION) \
+	CACHE_COMMAND="$(CACHE_COMMAND)" \
+	UPDATE_CACHE_COMMAND="$(UPDATE_CACHE_COMMAND)" \
+	TEST_SHA=$(TEST_SHA) \
+	PACKAGE_LOCATION=$(PWD)/output \
+	KONG_HOST=127.0.0.1 \
+	KONG_ADMIN_PORT=8444 \
+	KONG_PROXY_PORT=8000 \
+	KONG_TEST_CONTAINER_NAME=$(KONG_TEST_CONTAINER_NAME) \
+	KONG_ADMIN_URI="http://$(TEST_HOST):$(TEST_ADMIN_PORT)" \
+	KONG_PROXY_URI="http://$(TEST_HOST):$(TEST_PROXY_PORT)" \
+	TEST_COMPOSE_PATH=$(TEST_COMPOSE_PATH) \
+	KONG_BASE_IMAGE_NAME=$(KONG_BASE_IMAGE_NAME) \
 	./test/run_tests.sh
 endif
 
-run_tests:
-ifneq ($(RESTY_IMAGE_BASE),src)
-	cd test && \
-	$(CACHE_COMMAND) kong/kong-build-tools:test-runner-$(TEST_SHA) || \
-	docker build -t kong/kong-build-tools:test-runner-$(TEST_SHA) -f Dockerfile.test_runner .
-	cd test && \
-	docker run -t --network host -e RESTY_VERSION=$(RESTY_VERSION) -e KONG_VERSION=$(KONG_VERSION) -e ADMIN_URI=$(TEST_ADMIN_URI) -e PROXY_URI=$(TEST_PROXY_URI) kong/kong-build-tools:test-runner-$(TEST_SHA) /bin/bash -c "py.test -p no:logging -p no:warnings test_*.tavern.yaml"
-	-$(UPDATE_CACHE_COMMAND) kong/kong-build-tools:test-runner-$(TEST_SHA)
-endif
-
```

Comment:
it's pinned so this won't break but the idea here was
`test` -> setup the infrastructure, configure kong such that it can be tested
`run_tests` -> assumes the infra is setup and kong is configured all it does is run said tests

Giving the flexibility for other repositories to recycle the functional tests contained within this repository ( https://github.com/Kong/docker-kong/blob/master/tests.sh#L71 )

Reply:
Good point. Given now tests have the ability to individually start or stop a clean kong instance, I thought preserving the `test` target for triggering all the smoke tests would be okay. That'd also allow us to run all smoke tests (not only API functional tests) in docker-kong.